File: rust/src/models/component_override.rs
// this file is @generated
use serde::{Deserialize, Serialize};

use super::{price_component_id::PriceComponentId, price_entry::PriceEntry};

#[derive(Clone, Debug, Default, PartialEq, Deserialize, Serialize)]
pub struct ComponentOverride {
    pub component_id: PriceComponentId,

    pub name: String,

    pub price_entry: PriceEntry,
}

impl ComponentOverride {
    pub fn new(component_id: PriceComponentId, name: String, price_entry: PriceEntry) -> Self {
        Self {
            component_id,
            name,
            price_entry,
        }
    }
}

File: typescript/src/models/componentOverride.ts
// this file is @generated
import { type PriceComponentId, PriceComponentIdSerializer } from "./priceComponentId";
import { type PriceEntry, PriceEntrySerializer } from "./priceEntry";

export interface ComponentOverride {
  componentId: PriceComponentId;

  name: string;

  priceEntry: PriceEntry;
}

export const ComponentOverrideSerializer = {
  _fromJsonObject(object: any): ComponentOverride {
    return {
      componentId: PriceComponentIdSerializer._fromJsonObject(object["component_id"]),
      name: object["name"],
      priceEntry: PriceEntrySerializer._fromJsonObject(object["price_entry"]),
    };
  },

  _toJsonObject(self: ComponentOverride): any {
    return {
      component_id: PriceComponentIdSerializer._toJsonObject(self.componentId),
      name: self.name,
      price_entry: PriceEntrySerializer._toJsonObject(self.priceEntry),
    };
  },
};
